docs/tutorial/strlen.html

```diff
@@ -0,0 +1,103 @@
+* `wide(ptr)` constructor
+* `eve::load` function
+Here we are perfectly fine doing either of those until we decide to use our strlen with dynamic analysis.
+As was mentioned previously, it is OK to load from an aligned address but not from the view of C++.
+The dynamic analysis on this `strlen` will report an issue because you read outside of allocated memory.
+For `asan` and `tsan` we have special modifier `unsafe`. When we call `eve::unsafe(eve::load)(aligned_s)`,
+EVE detects build with these sanitizers and disable them for this load.
+`wide(ptr)` constructor does not have a way to do this.
+
+### `cur == zeroes`
+This is a pairwise comparison for equality between `cur` and `zero`.
+The result is `eve::logical<wide>` - a class representing N booleans where N is the width of our wide.
+
+### `eve::ignore_first ignore / eve::first_true[ignore]`
+`first_true` takes a `logical` and returns the optional position of the first `true` element.
+However we took a `previous_aligned_address` and not just `s` - so there might be some 0s we are not interested in. <br/>
+For this purpose `eve` provides some conditional mutators, most commonly used ones are:
+`ignore_first`, `ignore_last`, `ignore_extrema` (ignores from both sides).
+
+### `while (!match) / aligned_s += wide::static_size`
+While we didn't find the 0 - go to the next chunk. `wide::static_size` is the number of elements in the `wide`. Alternatively we could've used `wide.size()`.
+`static_size` has the advantage of being `constexpr` friendly.
+
+### `match = eve::first_true(cur == zeroes);`
+This is the same `first_true` but we no longer need to ignore anything.
+
+### `aligned_s.get() + *match - s`
+Finally - compute where we finished.
+
+# Comparison with real strlens
+
+The code above closely resembles [MacOs x86 strlen](https://opensource.apple.com/source/Libc/Libc-997.90.3/x86_64/string/strlen.s). <br/>
+With EVE compiled for `sse2`: [godbolt](https://godbolt.org/z/hvr3P5) <br/>
+The differences are minimal: clang handles the bit and address manipulations slightly differently.
+
+When comparing against [glibc arm-v8 (aarch64)](https://code.woboq.org/userspace/glibc/sysdeps/aarch64/multiarch/strlen_asimd.S.html) <br/>
+With eve we get: [eve, aarch64](https://godbolt.org/z/nz5eWe). <br/>
+The interesting differences are:
+* they compute position of the first match differently.
+* we compare with 0 and then look for `true` using `umaxv` instruction.
+  They rely on 0 being unsigned minimum and use `uminv` on the register directly.
+  Unfortunatly this trick only works for looking for 0 and we can look for arbitraty value.
+
+There are some substantial improvements one can make to this `strlen`, like unrolling which can be found in [glibc avx2 implementation](https://code.woboq.org/userspace/glibc/sysdeps/x86_64/multiarch/strlen-avx2.S.html).
+In the end - we are not advocating rewriting `strlen` with eve - this was just an example to show the interfaces <br/>
```

Comment:
I think the "we are not advocating" bit should actually be a note at the end of the intro.